src/omnisharp-http-utils.el

```diff
@@ -0,0 +1,39 @@
+
+(defun omnisharp-get-host ()
+  "Makes sure omnisharp-host is ended by / "
+  (if (string= (substring omnisharp-host -1 ) "/")
+      omnisharp-host
+    (concat omnisharp-host "/")))
+
+(defun omnisharp--get-api-url (api-name)
+  (concat (omnisharp-get-host) api-name))
+
+;;;###autoload
+(defun omnisharp-post-http-message (url &optional params)
+  "Post http request to server. Return result."
+  (let* ((response (omnisharp--submit-request url params))
+         (data (elt response 3)))
+    (unless data
+      (message "Error when sending request"))
+    data))
+
+(defun omnisharp--submit-request (url &optional params)
+  (require 'request)
+  (request (omnisharp--get-api-url url)
+           :type "POST"
+           :parser 'json-read
```

Comment:
Do you think there would be a benefit to using `omnisharp--json-read-from-string` here? I'm not sure how it would change the displaying of errors. Can you try it out with some simple call?